driver/controller.go

```diff
@@ -208,73 +202,98 @@ func (d *Driver) ControllerPublishVolume(ctx context.Context, req *csi.Controlle
+		result:    make(chan error),
+	}
+
+	d.attach <- ac
+	result := <-ac.result
+	close(ac.result)
+	if result == nil {
+		return controllerPublishVolumeSuccessResponse(fmt.Sprintf("disk-%d", diskID), req.NodeId, diskID), nil
+	}
+	return nil, result
+	// // check if volume exist before trying to attach it
+	// vol, err := d.client.Disks.Get(diskID)
+	// if err != nil {
+	// 	return nil, err
+	// }
+
+	// machineID, err := strconv.Atoi(req.NodeId)
+	// if err != nil {
+	// 	return nil, err
+	// }
+
+	// targetMachine, err := d.client.Machines.Get(machineID)
+	// if err != nil {
+	// 	return nil, err
+	// }
+
+	// machines, err := d.client.Machines.List(targetMachine.CloudspaceID)
+	// if err != nil {
+	// 	return nil, err
+	// }
+
+	// // detach it first if necessary
+	// for _, m := range *machines {
+	// 	if machineHasDisk(m, diskID) {
+	// 		if m.ID == targetMachine.ID {
+	// 			logger.Infof("Disk %d is already attached to machine %s. Nothing to do for us here.", diskID, targetMachine.Name)
+	// 			return controllerPublishVolumeSuccessResponse(vol.Name, req.NodeId, vol.ID)
+	// 		}
+	// 		logger.Infof("Disk %d is currently attached to machine %s. Detaching...", diskID, m.Name)
+	// 		attachConfig := &ovc.DiskAttachConfig{
+	// 			MachineID: m.ID,
+	// 			DiskID:    diskID,
+	// 		}
+	// 		d.client.Disks.Detach(attachConfig)
+	// 		logger.Infof("Disk %d is detached correctly from machine %s. Now attaching it to the target machine %s...", diskID, m.Name, targetMachine.Name)
+	// 		break
+	// 	}
+	// }
+
+	// // Attach the volume to the correct node
+	// attachConfig := &ovc.DiskAttachConfig{
+	// 	MachineID: targetMachine.ID,
+	// 	DiskID:    diskID,
+	// }
+	// err = d.client.Disks.Attach(attachConfig)
+	// if err != nil {
+	// 	return nil, err
+	// }
+	// logger.Infof("Disk %d has been attached correctly to target machine %s...", diskID, targetMachine.Name)
+	// return controllerPublishVolumeSuccessResponse(vol.Name, req.NodeId, vol.ID)
 }
 
-func controllerPublishVolumeSuccessResponse(volumeName, nodeID string, volumeID int) (*csi.ControllerPublishVolumeResponse, error) {
+func controllerPublishVolumeSuccessResponse(volumeName, nodeID string, volumeID int) *csi.ControllerPublishVolumeResponse {
 	return &csi.ControllerPublishVolumeResponse{
 		PublishContext: map[string]string{
 			"PublishInfoVolumeName": volumeName,
 			"PublishInfoVolumeID":   strconv.Itoa(volumeID),
 			"PublishInfoNodeID":     nodeID,
 		},
-	}, nil
+	}
 }
 
-// nodeHasDiskChecks if specified node has specified disk attached
-func nodeHasDisk(machine *ovc.MachineInfo, diskID int) bool {
+// machineHasDisk if specified node has specified disk attached
+func machineHasDisk(machine ovc.Machine, diskID int) bool {
 	for _, disk := range machine.Disks {
-		if disk.ID == diskID {
+		if disk == diskID {
 			return true
 		}
 	}
+	return false
+}
 
+// machineHasDisk2 if specified node has specified disk attached
+func machineHasDisk2(machine *ovc.MachineInfo, diskID int) bool {
```

Comment:
This shouldn't be needed? 
You should be able to dereference your `MachineInfo` pointer type when passing it onto `machineHasDisk` like `machineHasDisk(*myMachineInfo, myDiskID)`?
Also quickly looking around, I don't see this one used